### PR TITLE
Add more set methods

### PR DESCRIPTION
**Pull request by gsathya**
Fixes https://github.com/tc39/proposal-set-methods/issues/16

cc @rbuckton @mathiasbynens

### CODE REVIEW

File: spec/set-prototype-is-disjoint.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    1. If Type(_set_) is not Object, throw a *TypeError* exception.
+    1. If _set_ does not have a [[SetData]] internal slot, throw a *TypeError* exception.
+    1. Let _entries_ be the List that is _set_.[[SetData]].
+    1. If Type(_iterable_) is not Object, throw a *TypeError* exception.
+    1. If _iterable_ does not have [[SetData]] internal slot,
+        1. Let _otherSet_ be ? Construct(%Set%, _iterable_).
+    1. Else, let _otherSet_ be _iterable_.
+    1. Let _hasCheck_ be ? Get(_otherSet_, "has").
+    1. If IsCallable(_hasCheck_) is *false*, throw a *TypeError* exception.
+    1. For each _e_ that is an element of _entries_, do
```
Comment (mathiasbynens) [inline spec/set-prototype-is-disjoint.html]: Nit: `e` → `entry` (no reason to use single-letter variable names here)

Comment (mathiasbynens) [reply spec/set-prototype-is-disjoint.html]: Same goes for the other methods

File: spec/set-prototype-is-subset.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    1. If _set_ does not have a [[SetData]] internal slot, throw a *TypeError* exception.
+    1. Let _entries_ be the List that is _set_.[[SetData]].
+    1. If Type(_iterable_) is not Object, throw a *TypeError* exception.
+    1. If _iterable_ does not have [[SetData]] internal slot,
+        1. Let _otherSet_ be Construct(%Set%, _iterable_).
+    1. Else, let _otherSet_ be _iterable_.
+    1. Let _hasCheck_ be ? Get(_otherSet_, "has").
+    1. If IsCallable(_hasCheck_) is *false*, throw a *TypeError* exception.
+    1. For each _e_ that is an element of _entries_, do
+        1. Let _has_ be ? Call(_hasCheck_, _otherSet_, « e »)
```
Comment (mathiasbynens) [inline spec/set-prototype-is-subset.html]: Nit: missing `.`

File: spec/set-prototype-is-disjoint.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<p>When the `isDisjoint` method is called, the following steps are taken:</p>
```
Comment (mathiasbynens) [inline spec/set-prototype-is-disjoint.html]: Missing “with argument _…_” — see #27

File: docs/ecmarkup.js [hunk 4]
```diff
@@ -119,7 +119,11 @@ Search.prototype.search = function (searchString) {
     
     for (var i = 0; i < this.biblio.length; i++) {
       var entry = this.biblio[i];
-  
+      if (!entry.key) {
```
Comment (rbuckton) [inline docs/ecmarkup.js]: Is there a bug for this on https://github.com/bterlson/ecmarkup?

File: spec/set-prototype-is-subset.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<p>When the `isSubset` method is called, the following steps are taken:</p>
+
+<emu-alg>
+    1. Let _set_ be the *this* value.
+    1. If Type(_set_) is not Object, throw a *TypeError* exception.
+    1. If _set_ does not have a [[SetData]] internal slot, throw a *TypeError* exception.
+    1. Let _entries_ be the List that is _set_.[[SetData]].
+    1. If Type(_iterable_) is not Object, throw a *TypeError* exception.
+    1. If _iterable_ does not have [[SetData]] internal slot,
+        1. Let _otherSet_ be Construct(%Set%, _iterable_).
```
Comment (Ginden) [inline spec/set-prototype-is-subset.html]: Shouldn't we use SpeciesConstructor here?

File: spec/set-prototype-is-superset.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        1. Let _next_ be IteratorStep(_iter_).
+        1. If _next_ is *false*, return *true*.
+        1. Let _nextValue_ be ? IteratorValue(_next_).
+        1. Let _has_ be ? Call(_hasCheck_, _set_, « e »).
+        1. If _has_ is an abrupt completion, return ? IteratorClose(_iter_, _has_).
+        1. If _has_ is *false*, return *false*.
+
+</emu-alg>
+
+<p>The `length` property of the `isSuperSet` method is *1*.</p>
```
Comment (Ginden) [inline spec/set-prototype-is-superset.html]: Inconsistent casing.

File: spec/index.html [hunk 7]
```diff
@@ -39,3 +39,18 @@ <h1>Set.prototype.difference(_iterable_)</h1>
...
     <emu-import href="./set-prototype-symmetric-difference.html"></emu-import>
 </emu-clause>
+
+<emu-clause id="Set.prototype.isSubset">
+    <h1>Set.prototype.isSubset(_iterable_)</h1>
+    <emu-import href="./set-prototype-is-subset.html"></emu-import>
+</emu-clause>
+
+<emu-clause id="Set.prototype.isSuperset">
+    <h1>Set.prototype.isSuperset(_iterable_)</h1>
```
[the remark this answers is not in the document]
Comment (Ginden) [reply spec/index.html]: I agree that `isSupersetOf` and `isSubsetOf` are better names.

File: spec/set-prototype-is-disjoint.html [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+<emu-alg>
+    1. Let _set_ be the *this* value.
+    1. If Type(_set_) is not Object, throw a *TypeError* exception.
+    1. If _set_ does not have a [[SetData]] internal slot, throw a *TypeError* exception.
+    1. Let _entries_ be the List that is _set_.[[SetData]].
+    1. If Type(_iterable_) is not Object, throw a *TypeError* exception.
+    1. If _iterable_ does not have [[SetData]] internal slot,
+        1. Let _otherSet_ be ? Construct(%Set%, _iterable_).
```
Comment (Ginden) [inline spec/set-prototype-is-disjoint.html]: Shouldn't we use `SpeciesConstructor`?

File: spec/set-prototype-is-subset.html [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    1. Else, let _otherSet_ be _iterable_.
+    1. Let _hasCheck_ be ? Get(_otherSet_, "has").
+    1. If IsCallable(_hasCheck_) is *false*, throw a *TypeError* exception.
+    1. For each _e_ that is an element of _entries_, do
```
Comment (Ginden) [inline spec/set-prototype-is-subset.html]: It needs to check for `empty` entries.

File: spec/set-prototype-is-disjoint.html (hunk 8 above)
Comment (gsathya) [reply spec/set-prototype-is-disjoint.html]: Don't see any benefit here either

File: spec/set-prototype-is-subset.html (hunk 9 above)
Comment (gsathya) [reply spec/set-prototype-is-subset.html]: Why?

File: docs/ecmarkup.js (hunk 4 above)
Comment (gsathya) [reply docs/ecmarkup.js]: No. Why is this a bug? I'm not sure how ecmarkup generates this output, feel free to file an issue if you think this is incorrect.

File: spec/index.html [hunk 10]
```diff
@@ -39,3 +39,18 @@ <h1>Set.prototype.difference(_iterable_)</h1>
...
+    <h1>Set.prototype.isSubsetOf(_iterable_)</h1>
+    <emu-import href="./set-prototype-is-subset-of.html"></emu-import>
+</emu-clause>
+
+<emu-clause id="Set.prototype.isSupersetOf">
+    <h1>Set.prototype.isSupersetOf(_iterable_)</h1>
+    <emu-import href="./set-prototype-is-superset-of.html"></emu-import>
+</emu-clause>
+
+<emu-clause id="Set.prototype.isDisjointWith">
```
Comment (bakkot) [inline spec/index.html]: Small note: I believe "is disjoint from" is significantly more common than "is disjoint with". Googling each phrase gives almost an order of magnitude more results for "is disjoint from", and the first result for "is disjoint with" is someone asking if it's correct (whereas the first result for "is disjoint from" is a math article).

Comment (Ginden) [reply spec/index.html]: Can you raise new issue about this?

**Pull request merged**